src/prefect/telemetry/run_telemetry.py

```diff
@@ -0,0 +1,77 @@
+import time
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, Any, Dict, Optional
+
+from opentelemetry.trace import (
+    Status,
+    StatusCode,
+    get_tracer,
+)
+
+import prefect
+from prefect.client.schemas import TaskRun
+from prefect.client.schemas.objects import State
+
+if TYPE_CHECKING:
+    from opentelemetry.sdk.trace import Tracer
+
+
+@dataclass
+class RunTelemetry:
+    _tracer: "Tracer" = field(
+        default_factory=lambda: get_tracer("prefect", prefect.__version__)
+    )
+    _span = None
+
+    def start_span(
+        self,
+        task_run: TaskRun,
```

Comment:
I like how this is all encapsulated in a class. Could this be extended to work with flow runs too?

Reply:
It will be in a different PR.